stats/statsview.cpp

```diff
@@ -0,0 +1,982 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "statsview.h"
+#include "barseries.h"
+#include "boxseries.h"
+#include "legend.h"
+#include "pieseries.h"
+#include "scatterseries.h"
+#include "statsaxis.h"
+#include "statsstate.h"
+#include "statstranslations.h"
+#include "statsvariables.h"
+#include "core/divefilter.h"
+#include "core/subsurface-qt/divelistnotifier.h"
+#include <cmath>
+#include <QQuickItem>
+#include <QAbstractSeries>
+#include <QChart>
+#include <QGraphicsSceneHoverEvent>
+#include <QLocale>
+
+// Constants that control the graph layouts
+static const QColor quartileMarkerColor(Qt::red);
+static const double quartileMarkerSize = 15;
+
+static const QUrl urlStatsView = QUrl(QStringLiteral("qrc:/qml/statsview.qml"));
+
+// We use QtQuick's ChartView so that we can show the statistics on mobile.
+// However, accessing the ChartView from C++ is maliciously cumbersome and
+// the full QChart interface is not exported. Fortunately, the interface
+// leaks the QChart object: We can create a dummy-series and access the chart
+// object via the chart() accessor function. By creating a "PieSeries", the
+// ChartView does not automatically add axes.
```

Comment:
I hope that we aren't designing everything around an implementation detail (that leaked chart) that will disappear in the future...

Reply:
Hardly. Note that only two uses of the QtCharts module remain:
1) Drawing of the axes.
2) Connection to QML.
Both will be replaced in due course. I want to wait with the latter until the chart is included in mobile to avoid bad surprises.

Reply:
OK